ROADMAP.md

```diff
@@ -0,0 +1,287 @@
+let text = "Hello";
+
+TextInput::new(&mut text_input, &text, Message::TextChanged);
+```
+
+### TodoMVC example
+Once we have scrollables support and a text widget, we could build a [TodoMVC] example. It seems to be a very simple example and could showcase the API quite well.
+
+We could also consider releasing a `0.1.0` version at this point and share it as a milestone on different social platforms.
+
+[TodoMVC]: http://todomvc.com/
+
+### Multi-window support
+Open and control multiple windows at runtime.
+
+I think this could be achieved by implementing an additional trait in `iced_winit` similar to `Application` but with a slightly different `view` method, allowing users to control what is shown in each window.
+
+This approach should also allow us to perform custom optimizations for this particular use case.
+
+### Async actions
+Most applications need to perform work in the background, without freezing the UI while they work. The current architecture can be easily extended to achieve this.
+
+We can let users return an asynchronous action (i.e. a future) producing a `Message` in `Application::update`. The runtime will spawn each returned action in a thread pool and, once it finishes, feed the produced message back to `update`. This is also how [Elm] does it.
+
+When it comes to implementing this, [`winit`] already supports user-specific events that can be sent from another thread. Accomodating `iced_winit` for this functionality should not be too complicated.
+
+[Elm]: https://elm-lang.org/
+
+Here is an example of how the end-user API could look:
+
+```rust
+impl Application for WeatherReport {
+    fn update(&mut self, message: Message) -> Command<Message> {
+        match message {
+            Message::Refresh => {
+                self.state = State::Loading;
+
+                Command::from_future(
+                    Weather::request(self.location),
+                    Message::ReportReceived,
+                )
+            }
+            Message::ReportReceived(Ok(report)) => {
+                self.state = State::Show(report);
+
+                Command::none()
+            },
+            Message::ReportReceived(Err(error)) => {
+                self.state = State::Error(error);
+
+                Command::none()
+            }
+        }
+    }
+}
+```
+
+### Event subscriptions
+Besides performing async actions on demand, most applications also need to listen to events passively. An example of this could be a WebSocket connection, where messages can come in at any time.
+
+The idea here is to also follow [Elm]'s footsteps. We can add a method `subscriptions(&self) -> Subscription<Message>` to `Application` and keep the subscriptions alive in the runtime.
+
+The challenge here is designing the public API of subscriptions, so users can build their own, and _subscription diffing_, or basically detecting when a subscription is added/changed/removed. For this, we can take a look at the source code of [Elm] for inspiration.
```

Comment:
The flow of

> ... so users can build their own, and _subscription diffing_, or basically ...

made the message slightly unclear to me. Is the following the intended meaning?
```suggestion
The challenge here is designing the public API of subscriptions. Users should be able to create and subscribe to events, and we should be able to support _subscription diffing_, i.e. detecting when a subscription to an event is added, changed, or removed. For this, we can take a look at [Elm] for inspiration.
```

Reply:
Hmm... Close, but not exactly what I meant! A subscription could produce different kinds of events. What about something like this?

```suggestion
The challenge here is designing the public API of subscriptions. Ideally, users should be able to create their own and the GUI runtime should keep them alive while performing _subscription diffing_ (i.e. detecting when a subscription is added, changed, or removed). For this, we can take a look at [Elm] for inspiration.
```

In any case, I am aware the term "subscription" is a bit broad here. I am basically referring to Elm's subscriptions, which are similar to a background listener that can produce events concurrently.

Reply:
> What about something like this?

Sounds good! The clarification helps ^^